crates/modelardb_compression/src/models/timestamps.rs

```diff
@@ -327,17 +331,93 @@ mod tests {
             1579694400353,
             1579694400493,
             1579694400650,
-        ]);
+        ], Some(4));
+    }
+
+    #[test]
+    fn compress_and_decompress_bucket_sized_1_timestamps_for_an_irregular_time_series() {
+        compress_and_decompress_timestamps_for_a_time_series(&[
+            100,
+            100, // 0
+            200
+        ], Some(1));
+    }
+
+    #[test]
+    fn compress_and_decompress_bucket_sized_7_timestamps_for_an_irregular_time_series() {
+        compress_and_decompress_timestamps_for_a_time_series(&[
+            100,
+             37, // -63
+             38, //  64
+            200
+        ], Some(3));
+    }
+
+    #[test]
+    fn compress_and_decompress_bucket_sized_9_timestamps_for_an_irregular_time_series() {
+        compress_and_decompress_timestamps_for_a_time_series(&[
+            500,
+            245, // -255
+            246, //  256
+            500
+        ], Some(4));
+    }
+
+    #[test]
+    fn compress_and_decompress_fourth_bucket_12_sized_timestamps_for_an_irregular_time_series() {
```

Comment:
Very minor but is the naming of this test and the next consistent with the others? Should it be `sized_12` instead.